### ISSUE TITLE
config should use encoding.TextUnmarshaller on structs

### ISSUE THREAD

**Issue by alsamylkin** [uber-go/config#26]
Config populates values only for scalar types, functions and channel, but not for structs, right now it simply skips this check:
```
func TestHappyZapTextUnMarshallerParsing(t *testing.T) {
	t.Parallel()
	withYamlBytes([]byte(`level: debug`), func(provider Provider) {
		lvl := zap.NewAtomicLevel()
		err := provider.Get("level").Populate(&lvl)
		assert.NoError(t, err)
		assert.Equal(t, zap.DebugLevel, lvl.Level())
	})
}
```

Output:
```
Error:		Not equal: -1 (expected)
			!= 0 (actual)
```

**Comment by glibsm** [comment]
Is this now fixes through #27 ?

**Comment by alsamylkin** [comment]
Yes
